drivers/ps2/Kconfig.xec

```diff
@@ -19,6 +20,7 @@ config PS2_XEC_0
 
 config PS2_XEC_1
 	bool "PS2_XEC_1"
+	depends on SOC_SERIES_MEC1501X
```

Comment:
`PS2_XEC_0` and `PS2_XEC_1` both need to be removed

Reply:
MEC15xx has 2 PS2 controllers and MEC17xx has only one. Hence the` depends on SOC_SERIES_MEC1501X` is not applicable `PS2_XEC_0`

Reply:
The number of instances is defined in the SoC-level devicetrees, and after the refactoring in this PR to use `DT_INST_FOREACH_STATUS_OKAY`, the driver isn't using these Kconfig symbols anymore.

Reply:
removed `PS2_XEC_0 `and `PS2_XEC_1 `symbols